genesis_programs/src/lib.rs

```diff
@@ -21,43 +23,107 @@ extern crate solana_vest_program;
 #[macro_use]
 extern crate solana_vote_program;
 
-pub fn get() -> Vec<(String, Pubkey)> {
-    vec![
-        solana_system_program(),
-        solana_bpf_loader_program!(),
-        solana_budget_program!(),
-        solana_config_program!(),
-        solana_exchange_program!(),
-        #[cfg(feature = "move")]
-        solana_move_loader_program!(),
-        solana_stake_program!(),
-        solana_storage_program!(),
-        solana_vest_program!(),
-        solana_vote_program!(),
-    ]
+use log::*;
+use solana_runtime::bank::{Bank, EnteredEpochCallback};
+
+pub fn get(operating_mode: OperatingMode, epoch: Epoch) -> Option<Vec<(String, Pubkey)>> {
+    match operating_mode {
+        OperatingMode::Development => {
+            if epoch == 0 {
+                Some(vec![
+                    // Enable all SoftLaunch programs
+                    solana_system_program(),
+                    solana_bpf_loader_program!(),
+                    solana_config_program!(),
+                    solana_stake_program!(),
+                    solana_storage_program!(),
+                    solana_vest_program!(),
+                    solana_vote_program!(),
+                    // Programs that are only available in Development mode
+                    solana_budget_program!(),
+                    solana_exchange_program!(),
+                    #[cfg(feature = "move")]
+                    solana_move_loader_program!(),
+                ])
+            } else {
+                None
+            }
+        }
+        OperatingMode::SoftLaunch => {
+            if epoch == 0 {
+                // Voting and Staking only at epoch 0
+                Some(vec![
+                    solana_config_program!(),
+                    solana_stake_program!(),
+                    solana_vest_program!(),
```

Comment:
Let's kill `vest` and `config` for now. Anatoly thinks we can do without.

Reply:
Done